vizro-core/src/vizro/models/_navigation/navigation.py

```diff
@@ -1,38 +1,54 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Optional
+from typing import Optional
 
-from pydantic import PrivateAttr, validator
+from dash import html
+from pydantic import validator
 
 from vizro.models import VizroBaseModel
 from vizro.models._models_utils import _log_call
 from vizro.models._navigation._navigation_utils import _validate_pages
-from vizro.models.types import NavigationPagesType
-
-if TYPE_CHECKING:
-    from vizro.models._navigation._accordion import Accordion
+from vizro.models._navigation.accordion import Accordion
+from vizro.models._navigation.nav_bar import NavBar
+from vizro.models.types import NavPagesType, NavSelectorType
 
 
 class Navigation(VizroBaseModel):
     """Navigation in [`Dashboard`][vizro.models.Dashboard] to structure [`Pages`][vizro.models.Page].
 
     Args:
-        pages (Optional[NavigationPagesType]): See [`NavigationPagesType`][vizro.models.types.NavigationPagesType].
+        pages (Optional[NavPagesType]): See [`NavPagesType`][vizro.models.types.NavPagesType].
             Defaults to `None`.
+        selector (Optional[NavSelectorType]): See [`NavSelectorType`][vizro.models.types.NavSelectorType].
+            Defaults to `None`.)
     """
 
-    pages: Optional[NavigationPagesType] = None
-    _selector: Accordion = PrivateAttr()
+    pages: Optional[NavPagesType] = None
+    selector: Optional[NavSelectorType] = None
 
     # validators
-    _validate_pages = validator("pages", allow_reuse=True)(_validate_pages)
+    _validate_pages = validator("pages", allow_reuse=True, pre=True, always=True)(_validate_pages)
 
-    @_log_call
-    def pre_build(self):
-        from vizro.models._navigation._accordion import Accordion
+    @validator("selector", always=True)
+    def set_selector(cls, selector, values):
+        if selector is None:
+            return Accordion(pages=values.get("pages"))
 
-        self._selector = Accordion(pages=self.pages)  # type: ignore[arg-type]
+        if isinstance(selector, NavBar):
+            if selector.pages is None and selector.items is None:
+                selector.pages = values.get("pages")
+                selector.items = []
+                return selector
```

Comment:
I could be wrong, but I don't think we need the empty list instantiation of selector.items given that it's defaulting to an empty list if not provided?

```suggestion
            if selector.pages is None:
                selector.pages = values.get("pages")
```